packages/engine/src/output/utils.rs

```diff
@@ -2,14 +2,12 @@ use memory::shared_memory::MemoryId;
 
 use crate::{output::buffer::RELATIVE_PARTS_FOLDER, proto::ExperimentId};
 
-// TODO: move this to top-level
 #[derive(PartialEq, Eq, Clone)]
 pub enum EngineExitStatus {
     Success,
     Error,
 }
 
-// TODO: move this to top-level
 /// Shared memory cleanup in the process hard crash case.
 /// Not required for pod instances.
 pub fn cleanup_experiment(experiment_id: &ExperimentId, exit_status: EngineExitStatus) {
```

Comment:
This doesn't belong to `output` at all, `output` (besides that we want to remove the module) has nothing to do with the temporary files created by the python runner or the shared memory headers in `/dev/shm`. It's used to write output data into file, which should persist. I think it would be sensible to have this method on `ExperimentRunBase` (which is currently a POD, so maybe not a method but a free standing function). @Alfred-Mountfield may also have an idea on this.

Reply:
I don't like the idea of it being on `ExperimentRunBase`, that feels quite misleading. I think a free standing function in  `experiment/controller/mod.rs` or maybe `experiment/mod.rs` would be preferable

Reply:
`run.rs` already has the logic for starting experiments and such, so it feels natural to also have the clean-up

Reply:
Move `cleanup_experiment` to `run.rs` https://github.com/hashintel/hash/pull/494/commits/9489a7aa95ba8ebf8d568f6e3be826cbe15d2156